controllers/istiorevision/istiorevision_controller.go

```diff
@@ -494,6 +492,22 @@ func (r *Reconciler) mapPodToReconcileRequest(ctx context.Context, pod client.Ob
 	return nil
 }
 
+// ignoreStatusChange returns a predicate that ignores watch events where only the resource status changes; if
+// there are any other changes to the resource, the event is not ignored.
+// This ensures that the controller doesn't reconcile the entire IstioRevision every time the status of an owned
+// resource is updated. Without this predicate, the controller would continuously reconcile the IstioRevision
+// because the status.currentMetrics of the HorizontalPodAutoscaler object was updated.
+func ignoreStatusChange() predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.ObjectNew.GetGeneration() != e.ObjectOld.GetGeneration() || // this detects changes in the .spec
+				!reflect.DeepEqual(e.ObjectNew.GetLabels(), e.ObjectOld.GetLabels()) ||
+				!reflect.DeepEqual(e.ObjectNew.GetAnnotations(), e.ObjectOld.GetAnnotations()) ||
```

Comment:
Should we check for finalizers as well or is it not required?

Reply:
Yeah, good catch. Fixed.